tests/plugins/toolsets/test_kubernetes_yaml_injection.py

```
@@ -0,0 +1,263 @@
+    """Prove the canary harness detects a real leak: the pre-fix pattern
+    (kind interpolated into a double-quoted slot) MUST leak the canary."""
+    from holmes.core.tools import sanitize
+
+    os.environ[CANARY_ENV] = CANARY_VALUE
+    try:
+        # Mirrors the pre-fix error line: sanitize()d value inside "...'...'..."
+        vulnerable = (
+            'echo "Unable to find resource kind '
+            "'" + sanitize(f"$(printenv {CANARY_ENV})") + "'" '."'
+        )
+        result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)